packages/oui-modal-on-boarding/src/modal-on-boarding.controller.js

```diff
@@ -0,0 +1,209 @@
+            firstPanelElement.setAttribute("aria-hidden", "false");
+            const $firstPanelElement = angular.element(firstPanelElement);
+            $firstPanelElement.addClass("active");
+        }
+
+        if (this._isTouch) {
+            angular.element(this._componentElement).addClass("touchDevice");
+            this._setTouchEventListener();
+        }
+    }
+
+    $destroy () {
+        this.closeOnBoarding();
+    }
+
+    closeOnBoarding () {
+        if (this.onDismiss) {
+            this.onDismiss(event);
+        }
+
+        this.$document.off("keydown", evt => this._triggerKeyHandler(evt));
+        this.$scope.$broadcast("oui:modalOnBoarding:afterClose", this.id);
+    }
+
+    next () {
+        if (!this._isSliding) {
+            this._slide(Direction.NEXT);
+        }
+    }
+
+    prev () {
+        if (!this._isSliding) {
+            this._slide(Direction.PREV);
+        }
+    }
+
+    slideTo (index) {
+        const active = this.currentIndex;
+        const waiting = 10;
+
+        if (index > this.panels.length - 1 || index < 0 || index === active) {
+            return;
+        }
+
+        if (this._isSliding) {
+            this.$timeout(() => this.slideTo(index), waiting);
+            return;
+        }
+
+        this._slide(index > active ? Direction.NEXT : Direction.PREV, index);
+    }
+
+    _getIndexByDirection (direction) {
+        const isNextDirection = direction === Direction.NEXT;
+        const isPrevDirection = direction === Direction.PREV;
+        const lastItemIndex = this.panels.length - 1;
+        const isGoingToWrap = (isPrevDirection && this.currentIndex === 0) || (isNextDirection && this.currentIndex === lastItemIndex);
+
+        if (isGoingToWrap && !this.loop) {
+            return this.currentIndex;
+        }
+
+        const delta = direction === Direction.PREV ? -1 : 1;
+        const itemIndex = (this.currentIndex + delta) % this.panels.length;
+
+        return itemIndex === -1 ? this.panels.length - 1 : itemIndex;
+    }
+
+    _slide (direction, indexTo) {
+        const slideType = direction === Direction.NEXT ? Direction.LEFT : Direction.RIGHT;
+        const activePanel = this.panels[this.currentIndex].querySelector(`.${PANEL_CLASSNAME}`);
+        const $activePanel = angular.element(activePanel);
+        const nextPanelIndex = indexTo === 0 ? 0 : indexTo || (activePanel && this._getIndexByDirection(direction));
+        const nextPanel = this.panels[nextPanelIndex].querySelector(`.${PANEL_CLASSNAME}`);
+        const $nextPanel = angular.element(nextPanel);
+
+        if (nextPanel && $nextPanel.hasClass("active")) {
+            this._isSliding = false;
+            return;
+        }
+
+        if (!activePanel || !nextPanel) {
+            return;
+        }
+
+        this._isSliding = true;
+        const nextIsIllustrated = this._isIllustratedPanel(nextPanelIndex);
+
+        $nextPanel.addClass(direction);
```

Comment:
Same as $postLink.
This should be in your $timeout below

Reply:
DOM is sure all loaded at this time.
I don't think is required to do that !

Reply:
👍 I think it's not necessary. `$timeout` is useful when we make DOM manipulation in `$postLink` because we're not sure that DOM has been inserted yet. Here, the DOM is clearly ready.